mysql-test/t/percona_encrypt_tmp_files.test

```diff
@@ -1,5 +1,148 @@
 --source include/have_innodb.inc
 --source include/have_myisam.inc
+--source include/have_grep.inc
+--source include/have_debug_sync.inc
+
+#
+# Filesort temporary files
+#
+--let $encryption_mode = 0
+while($encryption_mode < 2)
+{
+  # Asserting expected values in the .opt file / provided via $restart_parameters
+  if($encryption_mode == 0)
+  {
+    --let $assert_text = Temporary file encryption must be enabled
+    --let $assert_cond = @@global.encrypt_tmp_files = 1
+    --source include/assert.inc
+    --let $assert_text = Sort buffer size must be 4M
+    --let $assert_cond = @@global.sort_buffer_size = 4*1048576
+    --source include/assert.inc
+  }
+  if($encryption_mode == 1)
+  {
+    --let $assert_text = Temporary file encryption must be disabled
+    --let $assert_cond = @@global.encrypt_tmp_files = 0
+    --source include/assert.inc
+    --let $assert_text = Sort buffer size must be 4M
+    --let $assert_cond = @@global.sort_buffer_size = 4*1048576
+    --source include/assert.inc
+  }
+  
+  # Determining server temp dir where 'filesort()' temporary files will be
+  # created.
+  --let $tmp_dir = `SELECT @@global.tmpdir`
+
+  # Determining server PID file name.
+  --let $pid_file = `SELECT @@global.pid_file`
+
+  # Creating a temporary table for reading ids from a file.
+  CREATE TEMPORARY TABLE id_table(id INT UNSIGNED);
+
+  # Reading the server PID from $pid_file
+  --replace_result $pid_file <pid_file>
+  --eval LOAD DATA LOCAL INFILE '$pid_file' INTO TABLE id_table
+  --let $pid_no = `SELECT id FROM id_table`
+  TRUNCATE TABLE id_table;
+
+  # Creating a simple table.
+  CREATE TABLE t1(
+    id BIGINT UNSIGNED NOT NULL PRIMARY KEY,
+    value CHAR(32) NOT NULL,
+    dummy VARCHAR(1024) NOT NULL
+  ) ENGINE=InnoDB;
+
+  # Filling the table with pseudo-random data based on MD5 hashes.
+  --let $length_multiplier = 16
+  eval INSERT INTO t1 VALUES (0, MD5(0), REPEAT(MD5(0), $length_multiplier));
+  --let $shift = 1
+  --let $n = 12
+  while($n)
+  {
+    eval INSERT INTO t1 SELECT $shift + id, MD5($shift + id), REPEAT(MD5($shift + id), $length_multiplier) FROM t1 ORDER BY id;
+    --let $shift = `SELECT $shift * 2`
+    --dec $n
+  }
+
+  # Establishing a new auxiliary connection.
+  --connect (con1,localhost,root,,)
+
+  # Making the server stop at 'after_find_all_keys' DEBUG_SYNC point
+  # (inside 'filesort()' after 'find_all_keys()')
+  # when executing the following 'SELECT sql_big_result' statement.
+  SET DEBUG_SYNC = 'after_find_all_keys SIGNAL after_find_all_keys_reached WAIT_FOR after_find_all_keys_signaled';
+  --send SELECT sql_big_result * FROM t1 ORDER BY dummy
+
+  # In the main connection waiting for the SELECT from the aux connection to
+  # reach 'after_find_all_keys' DEBUG_SYNC point and suspend.
+  --connection default
+  SET DEBUG_SYNC = 'now WAIT_FOR after_find_all_keys_reached';
+
+  # Looking for a file descriptor (symbolic link) from '/proc/$pid_no/fd'
```

Comment:
Whoa we are deep in --source include/linux.inc territory. Another argument for splitting the testcase, to let mac devs know locally if they broke something too badly